Automattic/wp-calypso
# Sites: redirect to onboarding flow when browsing the page with no sites

Related to pet6gk-xy-p2#comment-475.

## Proposed Changes

We can remove a point of friction from the onboarding flow if we simply redirect the user to `/setup/new-hosted-site` if they're browsing `/sites` with zero sites in their account. I added a link to `/setup/import-hosted-site` so that they can still import their sites if they want to.

![image](https://github.com/Automattic/wp-calypso/assets/26530524/7d3d04a8-71b3-4cf6-9cde-637aec548849)

If they are importing a site, they will also get the link to create a new one instead:

![image](https://github.com/Automattic/wp-calypso/assets/26530524/0adb308f-2c2d-47f9-b13c-173f72f73024)


"Why not redirect to the URL directly after signing up?"

This is a good question. I believe sending them to sites and then deciding what their flow should be would allow us to centralize the decision to a single place. But that's only me, I have no hard feelings about modifying the destination from `/start/hosting`. I was thinking about potential bounces too: if the user ends up browsing `/sites` in the future without going through the login step, they can start from where they left off providing they signed up as a developer.

## Testing Instructions

In an account with no sites, browse `/sites?hosting-flow=true` and you should be redirected to `/setup/new-hosted-site?hosting-flow=true`. Verify that "Import a site" link redirects you to `/setup/import-hosted-site?hosting-flow=true`.

You can browse `/sites` without the query parameter and verify that you'll see the regular onboarding flow if you're not a developer.
